tests/test_landing_pad_tracking.py

```diff
@@ -0,0 +1,146 @@
+        """        
+        detections1[1] = object_in_world.ObjectInWorld.create(0.5, 0.5, 4)[1]
+        success_flag, result = tracker.run(detections1)
+        print(detections1[2].spherical_variance, result.spherical_variance)
+        assert success_flag
+        assert result == detections1[2]
+        assert tracker._LandingPadTracking__unconfirmed_positives == [detections1[2], detections1[1], detections1[4], detections1[3]]
+
+    def test_run_multiple_inputs(self, tracker: landing_pad_tracking.LandingPadTracking, detections1: "list[object_in_world.ObjectInWorld]", detections2: "list[object_in_world.ObjectInWorld]"):
+        """
+        Test run with 2 inputs where some landing pads are similar
+        """
+        success_flag, result = tracker.run(detections1)
+        assert success_flag
+        assert result == detections1[2]
+        assert tracker._LandingPadTracking__unconfirmed_positives == [detections1[2], detections1[1], detections1[4], detections1[0], detections1[3]]
+
+        success_flag, result = tracker.run(detections2)
+        assert success_flag
+        assert result == detections2[0]
+        assert tracker._LandingPadTracking__unconfirmed_positives == [detections2[0], detections1[2], detections2[1], detections2[3], detections1[4], detections2[2], detections2[4], detections1[3]]
+
+    def test_run_with_confirmed_positive(self, tracker:landing_pad_tracking.LandingPadTracking, detections1: "list[object_in_world.ObjectInWorld]"):
+        """
+        Test run when there is a confirmed positive
+        """
+        confirmed_positive = object_in_world.ObjectInWorld.create(1, 1, 1)[1]
+        tracker.mark_confirmed_positive(confirmed_positive)
+        assert tracker._LandingPadTracking__confirmed_positives[0] == confirmed_positive
+        success_flag, result = tracker.run(detections1)
+        assert success_flag
+        assert result == confirmed_positive
+    
+    def test_mark_false_positive(self, tracker:landing_pad_tracking.LandingPadTracking, detections2: "list[object_in_world.ObjectInWorld]"):
+        """
+        Test if marking false positives removes similar landing pads
+        """
+        false_positive = object_in_world.ObjectInWorld.create(1, 1, 1)[1]
+        tracker.run(detections2)
+        tracker.mark_false_positive(false_positive)
+        assert tracker._LandingPadTracking__false_positives[0] == false_positive
+        assert tracker._LandingPadTracking__unconfirmed_positives == [detections2[3], detections2[2], detections2[4]]
+    
+    def test_run_with_false_positive(self, tracker:landing_pad_tracking.LandingPadTracking, detections2: "list[object_in_world.ObjectInWorld]"):
+        """
+        Test to see if run function doesn't add landing pads that are similar to false positives
+        """
+        false_positive = object_in_world.ObjectInWorld.create(1, 1, 1)[1]
+        tracker.mark_false_positive(false_positive)
```

Comment:
Avoid dependency on other class methods, modify class members directly instead as part of setup.